Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIBookmarks.mm

```diff
@@ -31,7 +31,111 @@
 #if ENABLE(WK_WEB_EXTENSIONS_BOOKMARKS)
 
 #import <WebKit/WKPreferencesPrivate.h>
+#import <WebKit/WKWebExtensionControllerDelegatePrivate.h>
 #import <WebKit/_WKFeature.h>
+#import <WebKit/_WKWebExtensionBookmarks.h>
+
+
+@interface _MockBookmarkNode : NSObject <_WKWebExtensionBookmark>
+- (instancetype)initWithDictionary:(NSDictionary *)dictionary;
+@property (nonatomic, strong) NSMutableDictionary *dictionary;
+@end
+
+@implementation _MockBookmarkNode
+- (instancetype)initWithDictionary:(NSDictionary *)dictionary
+{
+    self = [super init];
+    if (self)
+        _dictionary = [dictionary mutableCopy];
+    return self;
+}
+- (NSString *)identifierForWebExtensionContext:(WKWebExtensionContext *)context { return _dictionary[@"id"]; }
+- (NSString *)parentIdentifierForWebExtensionContext:(WKWebExtensionContext *)context { return _dictionary[@"parentId"]; }
+- (NSString *)titleForWebExtensionContext:(WKWebExtensionContext *)context { return _dictionary[@"title"]; }
+- (NSString *)urlForWebExtensionContext:(WKWebExtensionContext *)context { return _dictionary[@"url"]; }
+- (NSInteger)indexForWebExtensionContext:(WKWebExtensionContext *)context { return [_dictionary[@"index"] integerValue]; }
+- (NSArray<id<_WKWebExtensionBookmark>> *)childrenForWebExtensionContext:(WKWebExtensionContext *)context {
+    NSArray *childDictionaries = _dictionary[@"children"];
+    if (!childDictionaries)
+        return nil;
+    NSMutableArray<id<_WKWebExtensionBookmark>> *children = [NSMutableArray array];
+    for (NSDictionary *childDict in childDictionaries)
+        [children addObject:[[_MockBookmarkNode alloc] initWithDictionary:childDict]];
+    return children;
+}
```

Comment:
Nit: All of these methods should be on multiple lines.

```
- (NSString *)identifierForWebExtensionContext:(WKWebExtensionContext *)context
{ 
    return _dictionary[@"id"];
}
```

And should have the opening brace on the next line.